test/python/circuit/test_circuit_properties.py

```diff
@@ -616,6 +617,29 @@ def test_num_qubits_multiple_register_circuit(self):
         circ = QuantumCircuit(q_reg1, q_reg2, q_reg3)
         self.assertEqual(circ.num_qubits, 18)
 
+    def test_calibrations(self):
+        """Check if the calibrations provided are added correctly."""
+        circ = QuantumCircuit(2)
+        circ.rx(3.14, 0)
+        circ.ry(1.57, 1)
+
+        with pulse.build() as q0_x180:
+            pulse.play(pulse.library.Gaussian(20, 1.0, 3.0), pulse.DriveChannel(0))
+        with pulse.build() as q1_y90:
+            pulse.play(pulse.library.Gaussian(20, -1.0, 3.0), pulse.DriveChannel(1))
+
+        # Add calibration
+        circ.add_calibration(RXGate(3.14), [0], q0_x180)
+        circ.add_calibration(RYGate(1.57), [1], q1_y90)
+
+        self.assertEqual(set(circ.calibrations.keys()), {'rx', 'ry'})
+        self.assertEqual(set(circ.calibrations['rx'].keys()), {((0,), (3.14,))})
+        self.assertEqual(set(circ.calibrations['ry'].keys()), {((1,), (1.57,))})
+        self.assertEqual(circ.calibrations['rx'][((0,), (3.14,))].instructions,
+                         q0_x180.instructions)
+        self.assertEqual(circ.calibrations['ry'][((1,), (1.57,))].instructions,
+                         q1_y90.instructions)
+
```

Comment:
ideas for more:
  - custom cate
  - standard gate
  - with and without params
  - str gate name